.pre-commit-config.yaml

```diff
@@ -1,6 +1,11 @@
 # See https://pre-commit.com for more information
 # See https://pre-commit.com/hooks.html for more hooks
 
+# need to exclude to prevent circular imports
+exclude: |
+  (?x)^(
+    src/mercury_engine_data_structures/formats/__init__.py
+  )$
```

Comment:
Either the imports should (if possible) be reorganized that it's not a problem or just use `# ruff noqa: I001` at the top of the file instead of completly disable the pre commit.